reana-ui/src/pages/workflowList/WorkflowList.js

```diff
@@ -43,28 +47,51 @@ function Workflows() {
   const currentUTCTime = () => moment.utc().format("HH:mm:ss [UTC]");
   const [refreshedAt, setRefreshedAt] = useState(currentUTCTime());
   const [pagination, setPagination] = useState({ page: 1, size: PAGE_SIZE });
+  const [statusFilter, setStatusFilter] = useState([]);
+  const [searchFilter, setSearchFilter] = useState();
+  const [sortDir, setSortDir] = useState("desc");
   const dispatch = useDispatch();
   const config = useSelector(getConfig);
   const workflows = useSelector(getWorkflows);
   const workflowsCount = useSelector(getWorkflowsCount);
+  const hasUserWorkflows = useSelector(userHasWorkflows);
   const loading = useSelector(loadingWorkflows);
   const reanaToken = useSelector(getReanaToken);
   const interval = useRef(null);
 
   useEffect(() => {
-    dispatch(fetchWorkflows({ ...pagination }));
+    dispatch(
+      fetchWorkflows({ ...pagination }, searchFilter, statusFilter, sortDir)
+    );
 
     if (!interval.current && reanaToken) {
       interval.current = setInterval(() => {
-        dispatch(fetchWorkflows({ ...pagination }));
+        dispatch(
+          fetchWorkflows({ ...pagination }, searchFilter, statusFilter, sortDir)
+        );
         setRefreshedAt(currentUTCTime());
       }, config.poolingSecs * 1000);
     }
 
     return function cleanup() {
       clearInterval(interval.current);
     };
-  }, [config.poolingSecs, dispatch, pagination, reanaToken]);
+  }, [
+    config.poolingSecs,
+    dispatch,
+    pagination,
+    reanaToken,
+    searchFilter,
+    statusFilter,
+    sortDir,
+  ]);
+
+  const applyFilter = (filter) => (value) => {
+    unstable_batchedUpdates(() => {
+      filter(value);
+      setPagination({ ...pagination, page: 1 });
```

Comment:
Yes, exactly, I think we should always go back to page 1 after some filters are applied. Because there could be too few filtered results to paginate. Before I implemented this change it was very weird to use it from the UX perspective.